openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -0,0 +1,111 @@
+            default=False,
+            help='Submit tasks for processing.'
+        )
+
+    def handle(self, *args, **options):
+        programs_config = ProgramsApiConfig.current()
+        self.client = Client.objects.get(name=programs_config.OAUTH2_CLIENT_NAME)
+
+        if self.client.user is None:
+            msg = (
+                'No user is associated with the %s OAuth2 client. No tasks have been enqueued. '
+                'Associate a user with the client and try again.'
+            ).format(programs_config.OAUTH2_CLIENT_NAME)
+
+            raise CommandError(msg)
+
+        self._get_run_modes()
+
+        logger.info('Looking for users who may be eligible for a program certificate.')
+
+        self._get_usernames()
+
+        if options.get('commit'):
+            logger.info('Enqueuing tasks for %d eligible users.', len(self.usernames))
+        else:
+            logger.info(
+                'Found %d eligible users. To enqueue tasks, pass the -c or --commit flags.',
+                len(self.usernames)
+            )
+            return
+
+        succeeded, failed = 0, 0
+        for username in self.usernames:
+            try:
+                award_program_certificates.delay(username)
+            except:  # pylint: disable=bare-except
+                failed += 1
+                logger.exception('Failed to enqueue program certification task for user [%s]', username)
+            else:
+                succeeded += 1
+                logger.info('Successfully enqueued task for user [%s]', username)
+
+        logger.info('Done. %d succeeded. %d failed.', succeeded, failed)
+
+    def _get_run_modes(self):
+        """Find all run modes which are part of a program."""
+        programs = get_programs(self.client.user)
+        self.run_modes = self._flatten(programs)
+
+    def _flatten(self, programs):
+        """Flatten program dicts into a set of run modes."""
+        run_modes = set()
+        for program in programs:
+            for course_code in program['course_codes']:
+                for run in course_code['run_modes']:
+                    course_key = CourseKey.from_string(run['course_key'])
+                    run_modes.add(
+                        RunMode(course_key, run['mode_slug'])
+                    )
+
+        return run_modes
+
+    def _get_usernames(self):
+        """Identify a subset of users who may be eligible for a program certificate.
+
+        This is done by finding users who have earned a certificate in at least one
+        program course code's run mode.
+        """
+        query = reduce(
+            lambda x, y: x | y,
+            [Q(course_id=r.course_key, mode=r.mode_slug) for r in self.run_modes]
+        )
+
+        # TODO: Filter further, by passing status?
+        username_dicts = GeneratedCertificate.eligible_certificates.filter(query).values('user__username').distinct()
+        self.usernames = [d['user__username'] for d in username_dicts]
```

Comment:
Maybe print out to the console or log how many usernames have we retrieved with this method?